### PR TITLE
Add utility to verify metering reports [DPP-1105]

### PR DESCRIPTION
**Pull request by simonmaxen-da**
changelog_begin
changelog_end

### CODE REVIEW

**Review by skisel-da — APPROVED** [review]

File: ledger/participant-integration-api/BUILD.bazel [hunk 1]
```diff
@@ -435,3 +435,36 @@ da_scala_benchmark_jmh(
         "//libs-scala/contextualized-logging",
     ],
 )
+
+da_scala_library(
+    name = "metering-verification-app-lib",
```
Comment (mziolekda) [inline ledger/participant-integration-api/BUILD.bazel:440]: What would speak against promoting this target to a dedicated library at the `//ledger/metering-verification` level? It is rather unusual for an app to be buried so deep in the library code

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportKeySpec.scala [hunk 2]
```diff
@@ -25,7 +25,7 @@ class MeteringReportKeySpec extends AnyWordSpec with Matchers {
       EnterpriseKey(expected).key shouldBe expected
     }
     "read test key from test classpath" in {
-      val key = MeteringReportKey.readSystemResourceAsKey(
+      val key = MeteringReportKey.assertReadSystemResourceAsKey(
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportKeySpec.scala:28]: Can we add a test for invalid URL?

File: ledger/participant-integration-api/src/app/scala/com/daml/ledger/metering/Main.scala [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+    import scalaz._
+    import scalaz.syntax.traverse._
+    import std.either._
+    import std.list._
+
+    val dir = Path.of(keyDir)
+    if (Files.isDirectory(dir)) {
+      for {
+        keys <- Files.list(dir).toScala(List).traverse(readKey)
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/app/scala/com/daml/ledger/metering/Main.scala:101]: Wouldn't it be better to allow the folder to contain both key files as well as other files?

Comment (simonmaxen-da) [reply ledger/participant-integration-api/src/app/scala/com/daml/ledger/metering/Main.scala:101]: Only allowing key files is simplest:
- Directories are cheap
- We don't need to filter out files based on some file name pattern criteria.
- We can fail with an error if any file in the directory is not a valid key file.

File: ledger/participant-integration-api/BUILD.bazel (hunk 1 above)
Comment (simonmaxen-da) [reply ledger/participant-integration-api/BUILD.bazel:440]: Good idea.

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportKeySpec.scala (hunk 2 above)
Comment (simonmaxen-da) [reply ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportKeySpec.scala:28]: Done.

**Review by mziolekda — APPROVED** [review]

**Pull request merged**